clusterloader2/testing/network/config.yaml

```diff
@@ -0,0 +1,33 @@
+{{$PROTOCOL := .PROTOCOL}}
+{{$NUMBER_OF_SERVERS := .NUMBER_OF_SERVERS}}
+{{$NUMBER_OF_CLIENTS := .NUMBER_OF_CLIENTS}}
+
+name: network
+automanagedNamespaces: 1
+tuningSets:
+- name: Sequence
+  parallelismLimitedLoad:
+    parallelismLimit: 1
+steps:
+- name: start network metric measurement
+  measurements:
+    - Identifier: NetworkPerformanceMetrics
+      Method: NetworkPerformanceMetrics
+      Params:
+        action: start
+        duration: 10s
+        protocol: {{$PROTOCOL}}
+        numberOfServers: {{$NUMBER_OF_SERVERS}}
+        numberOfClients: {{$NUMBER_OF_CLIENTS}}
+- name: Wait for 25 secs controller to be up
```

Comment:
This isn't needed, right? The gather will wait until everything will finish anyway, right?

Reply:
yes its needed. There is no wait in gather implementation. When gather is executed , it will read the results stored in local cache and return.

Reply:
I see - I missed that during the review. We should fix that.

Basically gather should instead do:
- wait (with timeout) until all objects that we created have already been updated with the result

Reply:
There are certain aspects we need to think through:
a) Should we have wait based on number of pods configured ? or have a fixed timeout value ? If we have fixed timeout, at lower case runtime will be high. 
b) When timeout happens , it possible that say some pod didnot update result (due to error in pod or could be delay etc). Should we still go ahead with our percentile calculation and return the summary with whatever numbers we have or return error ? my opinion is , we should return whatever result is available.

Reply:
> Should we have wait based on number of pods configured ? or have a fixed timeout value ? If we have fixed timeout, at lower case runtime will be high.

You know what the expected "test duration" is (because you pass it to the workers).
So basically you should wait until (a) you receive results from all pods (b) but no longer than '"initial test delay" + "test duration" + headroom (let's say headroom being max(A (e.g. 10s), number of pods * B (e.g. 0.1s)))

> When timeout happens , it possible that say some pod didnot update result (due to error in pod or could be delay etc). Should we still go ahead with our percentile calculation and return the summary with whatever numbers we have or return error ? my opinion is , we should return whatever result is available.

yes - let's return whathever is available. But also log the pods from which we didn't got results.

Reply:
modified the gather() to wait for the response. Regarding logging , i think it will be hard to log about pods which didnot update the metrics. We may have to maintain an additional info about the pairs. and then display which have not sent the response.